query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryParser.scala

```diff
@@ -86,20 +86,34 @@ class QueryParser(analyzer: SoQLAnalyzer[SoQLType], maxRows: Option[Int], defaul
 
     val fuser = CompoundTypeFuser(fuseMap)
     val analyze: DatasetContext[SoQLType] => Seq[SoQLAnalysis[ColumnName, SoQLType]] = {
-      val utDsCtx = new UntypedDatasetContext {
-        override val columns: OrderedSet[ColumnName] = {
-          // exclude non-existing columns in the schema
-          val existedColumns = columnIdMapping.filter { case (k, v) => schema.contains(v) }
-          OrderedSet(existedColumns.keysIterator.toSeq: _*)
-        }
-      }
+
       val parsedStmts = new Parser().selectStatement(query)
+
+      // expand "select *" if it appears in the first statement only
+      val firstStmt = parsedStmts.head
       val expandedStmts = parsedStmts.map { parsed =>
-        val expanded = AliasAnalysis.expandSelection(parsed.selection)(utDsCtx)
-        parsed.copy(selection = Selection(None, None, expanded))
+        if (firstStmt.ne(parsed)) { parsed }
```

Comment:
Not sure I like this; it's sort of weird to map over a sequence but short-circuit for all but the first.  Might be better to do `val transformed = {...transformation of firstStmt}; parsedStatements.updated(0, transformed)`